core/src/main/java/org/opensearch/sql/analysis/TypeEnvironment.java

```diff
@@ -48,6 +50,12 @@ public TypeEnvironment(TypeEnvironment parent, SymbolTable symbolTable) {
   @Override
   public ExprType resolve(Symbol symbol) {
     for (TypeEnvironment cur = this; cur != null; cur = cur.parent) {
+      if (!cur.symbolTable.lookup(symbol).isEmpty()
```

Comment:
Looks like there is already a fallback mechanism [HERE](https://github.com/opensearch-project/sql/blob/main/core/src/main/java/org/opensearch/sql/analysis/ExpressionAnalyzer.java#L337-L339). Could we alter this to work for nested with partiql syntax? At a quick glance I'm not sure why it's not working.

Reply:
For some reason the running `SELECT message.info` doesn't hit that code actually. Will investigate. Thanks!